components/ontology-access/hgnc/api/src/main/java/org/phenotips/ontology/internal/GeneHGNCNomenclature.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.phenotips.ontology.internal;
+
+import org.phenotips.ontology.OntologyTerm;
+import org.phenotips.ontology.internal.solr.AbstractHGNCSolrOntologyService;
+import org.phenotips.ontology.internal.solr.SolrOntologyTerm;
+
+import org.xwiki.component.annotation.Component;
+import java.text.MessageFormat;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.LinkedHashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.regex.Pattern;
+
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.solr.client.solrj.util.ClientUtils;
+import org.apache.solr.common.SolrDocument;
+import org.apache.solr.common.params.CommonParams;
+import org.apache.solr.common.params.ModifiableSolrParams;
+import org.apache.solr.common.params.SolrParams;
+import org.apache.solr.common.params.SpellingParams;
+
+/**
+ * Provides access to the HUGO Gene Nomenclature Committee's GeneNames ontology. The ontology prefix is {@code HGNC}.
+ *
+ * @version $Id$
+ * @since 1.2RC1
+ */
+@Component
+@Named("hgnc")
+@Singleton
+public class GeneHGNCNomenclature extends AbstractHGNCSolrOntologyService
+{
+    /** For determining if a query is a an id. */
+    private static final Pattern ID_PATTERN = Pattern.compile("^HGNC:[0-9]+$", Pattern.CASE_INSENSITIVE);
+
+    @Override
+    public String getDefaultOntologyLocation()
+    {
+        return "http://www.genenames.org/cgi-bin/download?col=gd_hgnc_id&col=gd_app_sym&col=gd_app_name"
+            + "&col=gd_prev_sym&col=gd_aliases&col=gd_pub_acc_ids&col=gd_pub_eg_id&col=gd_pub_ensembl_id&col="
+            + "gd_pub_refseq_ids&col=family.id&col=family.name&col=md_eg_id&col=md_mim_id&col=md_refseq_id&col="
+            + "md_prot_id&col=md_ensembl_id&status=Approved&status_opt=2&where=&order_by=gd_app_sym_sort&format="
+            + "text&hgnc_dbtag=on&submit=submit";
```

Comment:
Most likely the URL looks like this because it's still work in progress. The parameters should of course be defined in static final fields, and the URL constructed from these, in a more maintainable manner.